proto/substrait/algebra.proto

```diff
@@ -605,6 +606,31 @@ message MergeJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// The nested loop join (NLJ) operator will hold the entire right input and iterate over it using the
+// left input, evaluating the join expression on the Cartesian product of all rows.
+message NestedLoopJoinRel {
+  RelCommon common = 1;
+  Rel left = 2;
+  Rel right = 3;
+  Expression expression = 4;
+
```

Comment:
Should you add post_join_filter?

Reply:
According to the existing NLJ definition, it doesn't require one. I think that makes sense because anything that goes in a post_join_filter could also just be included in the expression IIUC.

Reply:
Good point.  I believe you are correct.

Reply:
Although this same point would apply to the logical operator as well :thinking:

Reply:
Haha, interesting... 😅